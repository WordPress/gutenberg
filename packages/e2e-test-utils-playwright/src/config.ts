const {
	WP_USERNAME = 'admin',
	WP_PASSWORD = 'password',
	WP_BASE_URL = 'http://localhost:8889',
} = process.env;

const WP_ADMIN_USER = {
	username: WP_USERNAME,
	password: WP_PASSWORD,
} as const;

export { WP_ADMIN_USER, WP_USERNAME, WP_PASSWORD, WP_BASE_URL };
