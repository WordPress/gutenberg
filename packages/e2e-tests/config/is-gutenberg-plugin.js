// eslint-disable-next-line @wordpress/wp-global-usage
globalThis.IS_GUTENBERG_PLUGIN =
	process.env.npm_package_config_IS_GUTENBERG_PLUGIN === 'true';
