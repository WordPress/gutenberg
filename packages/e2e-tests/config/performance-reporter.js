const { readFileSync, existsSync } = require( 'fs' );

function average( array ) {
	return array.reduce( ( a, b ) => a + b ) / array.length;
}

class PerformanceReporter {
	onRunComplete() {
		const path = __dirname + '/../specs/results.json';

		if ( ! existsSync( path ) ) {
			return;
		}

		const results = readFileSync( path, 'utf8' );
		const { load, domcontentloaded, type } = JSON.parse( results );

		// eslint-disable-next-line no-console
		console.log( `
Average time to load: ${ average( load ) }ms
Average time to DOM content load: ${ average( domcontentloaded ) }ms
Average time to type character: ${ average( type ) }ms
Slowest time to type character: ${ Math.max( ...type ) }ms
Fastest time to type character: ${ Math.min( ...type ) }ms
		` );
	}
}

module.exports = PerformanceReporter;
