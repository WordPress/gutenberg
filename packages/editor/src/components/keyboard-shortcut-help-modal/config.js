/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const textFormattingShortcuts = [
	{
		keyCombination: { modifier: 'primary', character: 'b' },
		description: __( 'Make the selected text bold.' ),
	},
	{
		keyCombination: { modifier: 'primary', character: 'i' },
		description: __( 'Make the selected text italic.' ),
	},
	{
		keyCombination: { modifier: 'primary', character: 'k' },
		description: __( 'Convert the selected text into a link.' ),
	},
	{
		keyCombination: { modifier: 'primaryShift', character: 'k' },
		description: __( 'Remove a link.' ),
	},
	{
		keyCombination: { character: '[[' },
		description: __( 'Insert a link to a post or page.' ),
	},
	{
		keyCombination: { modifier: 'primary', character: 'u' },
		description: __( 'Underline the selected text.' ),
	},
	{
		keyCombination: { modifier: 'access', character: 'd' },
		description: __( 'Strikethrough the selected text.' ),
	},
	{
		keyCombination: { modifier: 'access', character: 'x' },
		description: __( 'Make the selected text inline code.' ),
	},
	{
		keyCombination: {
			modifier: 'access',
			character: '0',
		},
		aliases: [
			{
				modifier: 'access',
				character: '7',
			},
		],
		description: __( 'Convert the current heading to a paragraph.' ),
	},
	{
		keyCombination: { modifier: 'access', character: '1-6' },
		description: __(
			'Convert the current paragraph or heading to a heading of level 1 to 6.'
		),
	},
	{
		keyCombination: { modifier: 'primaryShift', character: 'SPACE' },
		description: __( 'Add non breaking space.' ),
	},
];
