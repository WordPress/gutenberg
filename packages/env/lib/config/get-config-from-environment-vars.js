'use strict';
/**
 * Internal dependencies
 */
const {
	parseSourceString,
	includeTestsPath,
} = require( './parse-source-string' );
const { checkPort, checkVersion, checkString } = require( './validate-config' );

/**
 * @typedef {import('./parse-source-string').WPSource} WPSource
 */

/**
 * Environment variable configuration.
 *
 * @typedef WPEnvironmentVariableConfig
 * @property {?number}                  port             An override for the development environment's port.
 * @property {?number}                  mysqlPort        An override for the development environment's MySQL port.
 * @property {?number}                  testsPort        An override for the testing environment's port.
 * @property {?number}                  testsMysqlPort   An override for the testing environment's MySQL port.
 * @property {?WPSource}                coreSource       An override for all environment's coreSource.
 * @property {?string}                  phpVersion       An override for all environment's PHP version.
 * @property {?Object.<string, string>} lifecycleScripts An override for various lifecycle scripts.
 */

/**
 * Gets configuration options from environment variables.
 *
 * @param {string} cacheDirectoryPath Path to the work directory located in ~/.wp-env.
 *
 * @return {WPEnvironmentVariableConfig} Any configuration options parsed from the environment variables.
 */
module.exports = function getConfigFromEnvironmentVars( cacheDirectoryPath ) {
	const environmentConfig = {
		port: getPortFromEnvironmentVariable( 'WP_ENV_PORT' ),
		mysqlPort: getPortFromEnvironmentVariable( 'WP_ENV_MYSQL_PORT' ),
		testsPort: getPortFromEnvironmentVariable( 'WP_ENV_TESTS_PORT' ),
		testsMysqlPort: getPortFromEnvironmentVariable(
			'WP_ENV_TESTS_MYSQL_PORT'
		),
		lifecycleScripts: getLifecycleScriptOverrides(),
	};

	if ( process.env.WP_ENV_CORE ) {
		environmentConfig.coreSource = includeTestsPath(
			parseSourceString( process.env.WP_ENV_CORE, {
				cacheDirectoryPath,
			} ),
			{ cacheDirectoryPath }
		);
	}

	if ( process.env.WP_ENV_PHP_VERSION ) {
		checkVersion(
			'environment variable',
			'WP_ENV_PHP_VERSION',
			process.env.WP_ENV_PHP_VERSION
		);
		environmentConfig.phpVersion = process.env.WP_ENV_PHP_VERSION;
	}

	return environmentConfig;
};

/**
 * Parses an environment variable which should be a port.
 *
 * @param {string} varName The environment variable to check (e.g. WP_ENV_PORT).
 *
 * @return {number} The parsed port number.
 */
function getPortFromEnvironmentVariable( varName ) {
	if ( ! process.env[ varName ] ) {
		return undefined;
	}

	const port = parseInt( process.env[ varName ] );

	// Throw an error if it is not parseable as a number.
	checkPort( 'environment variable', varName, port );

	return port;
}

/**
 * Parses the lifecycle script environment variables.
 *
 * @return {Object.<string, string>} The parsed lifecycle scripts.
 */
function getLifecycleScriptOverrides() {
	const lifecycleScripts = {};

	// Find all of the lifecycle script overrides and parse them.
	const lifecycleEnvironmentVars = {
		WP_ENV_LIFECYCLE_SCRIPT_AFTER_START: 'afterStart',
		WP_ENV_LIFECYCLE_SCRIPT_AFTER_CLEAN: 'afterClean',
		WP_ENV_LIFECYCLE_SCRIPT_AFTER_DESTROY: 'afterDestroy',
	};
	for ( const envVar in lifecycleEnvironmentVars ) {
		const scriptValue = process.env[ envVar ];
		if ( scriptValue === undefined ) {
			continue;
		}

		checkString( 'environment variable', envVar, scriptValue );
		lifecycleScripts[ lifecycleEnvironmentVars[ envVar ] ] = scriptValue;
	}

	return lifecycleScripts;
}
