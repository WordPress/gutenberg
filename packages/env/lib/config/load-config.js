'use strict';
/**
 * External dependencies
 */
const path = require( 'path' );
const fs = require( 'fs' ).promises;

/**
 * Internal dependencies
 */
const getCacheDirectory = require( './get-cache-directory' );
const md5 = require( '../md5' );
const { parseConfig, getConfigFilePath } = require( './parse-config' );
const postProcessConfig = require( './post-process-config' );

/**
 * @typedef {import('./parse-config').WPRootConfig} WPRootConfig
 * @typedef {import('./parse-config').WPEnvironmentConfig} WPEnvironmentConfig
 */

/**
 * wp-env configuration.
 *
 * @typedef WPConfig
 * @property {string}                               name                    Name of the environment.
 * @property {string}                               configDirectoryPath     Path to the .wp-env.json file.
 * @property {string}                               workDirectoryPath       Path to the work directory located in ~/.wp-env.
 * @property {string}                               dockerComposeConfigPath Path to the docker-compose.yml file.
 * @property {boolean}                              detectedLocalConfig     If true, wp-env detected local config and used it.
 * @property {Object.<string, string>}              lifecycleScripts        Any lifecycle scripts that we might need to execute.
 * @property {Object.<string, WPEnvironmentConfig>} env                     Specific config for different environments.
 * @property {boolean}                              debug                   True if debug mode is enabled.
 * @property {boolean}                              disablePortMapping      True if the container ports should not be exposed to the host.
 */

/**
 * Loads any configuration from a given directory.
 *
 * @param {string} configDirectoryPath The directory we want to load the config from.
 *
 * @return {WPConfig} The config object we've loaded.
 */
module.exports = async function loadConfig( configDirectoryPath ) {
	const configFilePath = getConfigFilePath( configDirectoryPath );

	const cacheDirectoryPath = path.resolve(
		await getCacheDirectory(),
		md5( configFilePath )
	);

	// Parse any configuration we found in the given directory.
	// This comes merged and prepared for internal consumption.
	let config = await parseConfig( configDirectoryPath, cacheDirectoryPath );

	// Make sure to perform any additional post-processing that
	// may be needed before the config object is ready for
	// consumption elsewhere in the tool.
	config = postProcessConfig( config );

	return {
		name: path.basename( configDirectoryPath ),
		dockerComposeConfigPath: path.resolve(
			cacheDirectoryPath,
			'docker-compose.yml'
		),
		configDirectoryPath,
		workDirectoryPath: cacheDirectoryPath,
		detectedLocalConfig: await hasLocalConfig( [
			configFilePath,
			getConfigFilePath( configDirectoryPath, 'override' ),
		] ),
		lifecycleScripts: config.lifecycleScripts,
		env: config.env,
	};
};

/**
 * Checks to see whether or not there is any configuration present in the directory.
 *
 * @param {string[]} configFilePaths The config files we want to check for existence.
 *
 * @return {Promise<boolean>} A promise indicating whether or not a local config is present.
 */
async function hasLocalConfig( configFilePaths ) {
	for ( const filePath of configFilePaths ) {
		try {
			await fs.stat( filePath );
			return true;
		} catch {}
	}

	return false;
}
