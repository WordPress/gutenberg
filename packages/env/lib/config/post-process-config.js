'use strict';
/**
 * Internal dependencies
 */
const mergeConfigs = require( './merge-configs' );
const addOrReplacePort = require( './add-or-replace-port' );
const { ValidationError } = require( './validate-config' );

/**
 * External dependencies
 */
const fs = require( 'fs' );
const devcert = require( 'devcert' );

/**
 * @typedef {import('./parse-config').WPRootConfig} WPRootConfig
 * @typedef {import('./parse-config').WPEnvironmentConfig} WPEnvironmentConfig
 */

/**
 * Performs any additional post-processing on the config object.
 *
 * @param {WPEnvironmentConfig} config The config to process.
 *
 * @return {WPEnvironmentConfig} The config after post-processing.
 */
module.exports = async function postProcessConfig( config ) {
	config = await enableHttps( config );

	// Make sure that we're operating on a config object that has
	// complete environment configs for convenience.
	config = mergeRootToEnvironments( config );

	config = appendPortToWPConfigs( config );

	validate( config );
	return config;
};

/**
 * Merges the root config and each environment together in order to make sure each environment has
 * a full config object to work with internally. This makes it easier than having to try and
 * resolve the full config options every time we want to use them for something.
 *
 * @param {WPEnvironmentConfig} config The config to process.
 *
 * @return {WPRootConfig} The config object with the root options merged together with the environment-specific options.
 */
function mergeRootToEnvironments( config ) {
	// Some root-level options need to be handled early because they have a special
	// cascade behavior that would break the normal merge. After merging we then
	// delete them to avoid that breakage and add them back before we return.
	const removedRootOptions = {};
	if (
		config.port !== undefined &&
		config.env.development.port === undefined
	) {
		removedRootOptions.port = config.port;
		config.env.development.port = config.port;
		delete config.port;
	}
	if (
		config.testsPort !== undefined &&
		config.env.tests.port === undefined
	) {
		removedRootOptions.testsPort = config.testsPort;
		config.env.tests.port = config.testsPort;
		delete config.testsPort;
	}
	if ( config.lifecycleScripts !== undefined ) {
		removedRootOptions.lifecycleScripts = config.lifecycleScripts;
		delete config.lifecycleScripts;
	}
	if ( config.https !== undefined ) {
		removedRootOptions.https = config.https;
		delete config.https;
	}
	if ( config.sslCertPath !== undefined ) {
		removedRootOptions.sslCertPath = config.sslCertPath;
		delete config.sslCertPath;
	}
	if ( config.sslKeyPath !== undefined ) {
		removedRootOptions.sslKeyPath = config.sslKeyPath;
		delete config.sslKeyPath;
	}
	if (
		config.httpsPort !== undefined &&
		config.env.development.httpsPort === undefined
	) {
		removedRootOptions.httpsPort = config.httpsPort;
		config.env.development.httpsPort = config.httpsPort;
		delete config.httpsPort;
	}
	if (
		config.testsHttpsPort !== undefined &&
		config.env.tests.httpsPort === undefined
	) {
		removedRootOptions.testsHttpsPort = config.testsHttpsPort;
		config.env.tests.httpsPort = config.testsHttpsPort;
		delete config.testsHttpsPort;
	}

	// Merge the root config and the environment configs together so that
	// we can ignore the root config and have full environment configs.
	for ( const env in config.env ) {
		config.env[ env ] = mergeConfigs(
			deepCopyRootOptions( config ),
			config.env[ env ]
		);
	}

	// Set any root-level options we reset back.
	for ( const option in removedRootOptions ) {
		config[ option ] = removedRootOptions[ option ];
	}

	return config;
}

/**
 * Appends the configured port to certain wp-config options.
 *
 * @param {WPRootConfig} config The config to process.
 *
 * @return {WPRootConfig} The config after post-processing.
 */
function appendPortToWPConfigs( config ) {
	const options = [ 'WP_TESTS_DOMAIN', 'WP_SITEURL', 'WP_HOME' ];

	// We are only interested in editing the config options for environment-specific configs.
	// If we made this change to the root config it would cause problems since they would
	// be mapped to all environments even though the ports will be different.
	for ( const env in config.env ) {
		// There's nothing to do without any wp-config options set.
		if ( config.env[ env ].config === undefined ) {
			continue;
		}

		if ( config.env[ env ].port === undefined ) {
			continue;
		}

		// Make sure that the port is on the option if it's present.
		for ( const option of options ) {
			if ( config.env[ env ].config[ option ] === undefined ) {
				continue;
			}

			config.env[ env ].config[ option ] = addOrReplacePort(
				config.env[ env ].config[ option ],
				config.env[ env ].httpsPort ?? config.env[ env ].port,
				// Don't replace the port if one is already set on WP_HOME.
				option !== 'WP_HOME'
			);
		}
	}

	return config;
}

async function enableHttps( config ) {
	if ( config.https === true ) {
		const workDirectoryPath = config.coreSource.path.replace(
			/\/WordPress$/,
			''
		);
		const httpsUrl = ( url ) => url.replace( /^http:\/\//i, 'https://' );
		const domainName = config.config.WP_HOME.replace( /^https?:\/\//i, '' );

		// change the siteurl and home to https
		config.config.WP_SITEURL = httpsUrl( config.config.WP_SITEURL );
		config.config.WP_HOME = httpsUrl( config.config.WP_HOME );

		// check if certificate and key are set, if not we will create them
		if (
			config.sslCertPath === undefined &&
			config.sslKeyPath === undefined
		) {
			const certsDir = workDirectoryPath + '/certs';
			const certFile = `${ certsDir }/${ domainName }.crt`;
			const keyFile = `${ certsDir }/${ domainName }.key`;
			await devcert
				.certificateFor( domainName )
				.then( ( { key, cert } ) => {
					if ( ! fs.existsSync( certsDir ) ) {
						fs.mkdirSync( certsDir );
					}

					if (
						fs.existsSync( certFile ) &&
						fs.existsSync( keyFile )
					) {
						return;
					}

					fs.writeFileSync( certFile, cert );
					fs.writeFileSync( keyFile, key );
				} );

			config.sslCertPath = certFile;
			config.sslKeyPath = keyFile;
		} else {
			throw new ValidationError(
				'Invalid config: sslCertPath and sslKeyPath must both be set.'
			);
		}
	}

	return config;
}

/**
 * Examines the config to make sure that none of the environments share the same port.
 *
 * @param {WPRootConfig} config The config to process.
 */
function validatePortUniqueness( config ) {
	// We're going to build a map of the environments and their port
	// so we can accomodate root-level config options more easily.
	const environmentPorts = {};

	// Add all of the environments to the map. This will
	// overwrite any root-level options if necessary.
	for ( const env in config.env ) {
		if ( config.env[ env ].port === undefined ) {
			throw new ValidationError(
				`The "${ env }" environment has an invalid port.`
			);
		}

		environmentPorts[ env ] = config.env[ env ].port;
	}

	// This search isn't very performant, but, we won't ever be
	// checking more than a few entries so it doesn't matter.
	for ( const env in environmentPorts ) {
		for ( const check in environmentPorts ) {
			if ( env === check ) {
				continue;
			}

			if ( environmentPorts[ env ] === environmentPorts[ check ] ) {
				throw new ValidationError(
					`The "${ env }" and "${ check }" environments may not have the same port.`
				);
			}
		}
	}
}

/**
 * Perform any validation that can only happen after post-processing has occurred.
 *
 * @param {WPRootConfig} config The config to validate.
 */
function validate( config ) {
	validatePortUniqueness( config );
}

/**
 * Creates a deep copy of the root options in the config object that we can use to avoid
 * accidentally sharing object state between different environments.
 *
 * @param {WPRootConfig} config The root config object to copy.
 *
 * @return {WPRootConfig} A deep copy of the root config object.
 */
function deepCopyRootOptions( config ) {
	// Create a shallow clone of the object first so we can operate on it safetly.
	const rootConfig = Object.assign( {}, config );

	// Since we're only dealing with the root options we don't want the environments.
	delete rootConfig.env;

	if ( rootConfig.config !== undefined ) {
		rootConfig.config = Object.assign( {}, rootConfig.config );
	}
	if ( rootConfig.mappings !== undefined ) {
		rootConfig.mappings = Object.assign( {}, rootConfig.mappings );
	}
	if ( rootConfig.pluginSources !== undefined ) {
		rootConfig.pluginSources = [ ...rootConfig.pluginSources ];
	}
	if ( rootConfig.themeSources !== undefined ) {
		rootConfig.themeSources = [ ...rootConfig.themeSources ];
	}

	return rootConfig;
}
