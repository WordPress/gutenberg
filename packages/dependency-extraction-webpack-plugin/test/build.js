/**
 * External dependencies
 */
const fs = require( 'fs' );
const glob = require( 'glob' ).sync;
const mkdirp = require( 'mkdirp' ).mkdirp.sync;
const path = require( 'path' );
const rimraf = require( 'rimraf' ).sync;
const webpack = require( 'webpack' );

const fixturesPath = path.join( __dirname, 'fixtures' );
const configFixtures = fs.readdirSync( fixturesPath ).sort();

afterAll( () => rimraf( path.join( __dirname, 'build' ) ) );

describe.each( /** @type {const} */ ( [ 'scripts', 'modules' ] ) )(
	'DependencyExtractionWebpackPlugin %s',
	( moduleMode ) => {
		describe.each( configFixtures )( 'Webpack `%s`', ( configCase ) => {
			const testDirectory = path.join( fixturesPath, configCase );
			const outputDirectory = path.join(
				__dirname,
				'build',
				moduleMode,
				configCase
			);

			beforeEach( () => {
				rimraf( outputDirectory );
				mkdirp( outputDirectory );
			} );

			// This afterEach is necessary to prevent watched tests from retriggering on every run.
			afterEach( () => rimraf( outputDirectory ) );

			test( 'should produce expected output', async () => {
				const options = Object.assign(
					{
						name: `${ configCase }-${ moduleMode }`,
						target: 'web',
						context: testDirectory,
						entry: './index.js',
						mode: 'production',
						optimization: {
							minimize: false,
							chunkIds: 'named',
							moduleIds: 'named',
						},
						output: {},
						experiments: {},
					},
					require( path.join( testDirectory, 'webpack.config.js' ) )
				);
				options.output.path = outputDirectory;

				if ( moduleMode === 'modules' ) {
					options.target = 'es2024';
					options.output.module = true;
					options.output.chunkFormat = 'module';
					options.output.library = options.output.library || {};
					options.output.library.type = 'module';
					options.experiments.outputModule = true;
				}

				/** @type {webpack.Stats} */
				const stats = await new Promise( ( resolve, reject ) =>
					webpack( options, ( err, _stats ) => {
						if ( err ) {
							return reject( err );
						}
						resolve( _stats );
					} )
				);

				/* eslint-disable jest/no-conditional-expect */
				if ( configCase.includes( 'error' ) ) {
					expect( stats.hasErrors() ).toBe( true );
					expect(
						stats.toString( { errors: true, all: false } )
					).toMatchSnapshot();
					return;
				}
				/* eslint-enable jest/no-conditional-expect */

				if ( stats.hasErrors() ) {
					throw new Error(
						stats.toString( { errors: true, all: false } )
					);
				}

				const assetFiles = glob(
					`${ outputDirectory }/+(*.asset|assets).@(json|php)`
				);

				expect( assetFiles.length ).toBeGreaterThan( 0 );

				// Asset files should match.
				assetFiles.forEach( ( assetFile ) => {
					const assetBasename = path.basename( assetFile );

					expect(
						fs.readFileSync( assetFile, 'utf-8' )
					).toMatchSnapshot(
						`Asset file '${ assetBasename }' should match snapshot`
					);
				} );

				const compareByModuleIdentifier = ( m1, m2 ) => {
					const i1 = m1.identifier();
					const i2 = m2.identifier();
					if ( i1 < i2 ) {
						return -1;
					}
					if ( i1 > i2 ) {
						return 1;
					}
					return 0;
				};

				// Webpack stats external modules should match.
				const externalModules = Array.from( stats.compilation.modules )
					.filter( ( { externalType } ) => externalType )
					.sort( compareByModuleIdentifier )
					.map( ( module ) => ( {
						externalType: module.externalType,
						request: module.request,
						userRequest: module.userRequest,
					} ) );
				expect( externalModules ).toMatchSnapshot(
					'External modules should match snapshot'
				);
			} );
		} );
	}
);
