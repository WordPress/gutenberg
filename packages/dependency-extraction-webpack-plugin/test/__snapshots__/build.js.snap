// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DependencyExtractionWebpackPlugin Webpack \`combine-assets\` should produce expected output: Asset file 'assets.php' should match snapshot 1`] = `
"<?php return array('fileA.js' => array('dependencies' => array('lodash', 'wp-blob'), 'version' => '52d35a0af8a5fb869915'), 'fileB.js' => array('dependencies' => array('wp-token-list'), 'version' => '0af6c51a8e6ac934b85a'));
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`combine-assets\` should produce expected output: External modules should match snapshot 1`] = `
[
  {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
  {
    "externalType": "window",
    "request": [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
  {
    "externalType": "window",
    "request": [
      "wp",
      "tokenList",
    ],
    "userRequest": "@wordpress/token-list",
  },
]
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`dynamic-import\` should produce expected output: Asset file 'main.asset.php' should match snapshot 1`] = `
"<?php return array('dependencies' => array('wp-blob'), 'version' => '47f8a22aa6c04c930f5f');
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`dynamic-import\` should produce expected output: External modules should match snapshot 1`] = `
[
  {
    "externalType": "window",
    "request": [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
]
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`function-output-filename\` should produce expected output: Asset file 'chunk--main--main.asset.php' should match snapshot 1`] = `
"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => '35d057cb1c8a30094445');
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`function-output-filename\` should produce expected output: External modules should match snapshot 1`] = `
[
  {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
  {
    "externalType": "window",
    "request": [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
]
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`has-extension-suffix\` should produce expected output: Asset file 'index.min.asset.php' should match snapshot 1`] = `
"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => '27e49c80f699647ec11c');
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`has-extension-suffix\` should produce expected output: External modules should match snapshot 1`] = `
[
  {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
  {
    "externalType": "window",
    "request": [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
]
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`no-default\` should produce expected output: Asset file 'main.asset.php' should match snapshot 1`] = `
"<?php return array('dependencies' => array(), 'version' => 'cdc72dbcdf872f6141c7');
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`no-default\` should produce expected output: External modules should match snapshot 1`] = `[]`;

exports[`DependencyExtractionWebpackPlugin Webpack \`no-deps\` should produce expected output: Asset file 'main.asset.php' should match snapshot 1`] = `
"<?php return array('dependencies' => array(), 'version' => '091ffcd70d94dd16e773');
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`no-deps\` should produce expected output: External modules should match snapshot 1`] = `[]`;

exports[`DependencyExtractionWebpackPlugin Webpack \`option-function-output-filename\` should produce expected output: Asset file 'chunk--main--main.asset.php' should match snapshot 1`] = `
"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => '35d057cb1c8a30094445');
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`option-function-output-filename\` should produce expected output: External modules should match snapshot 1`] = `
[
  {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
  {
    "externalType": "window",
    "request": [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
]
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`option-output-filename\` should produce expected output: Asset file 'main-foo.asset.php' should match snapshot 1`] = `
"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => '35d057cb1c8a30094445');
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`option-output-filename\` should produce expected output: External modules should match snapshot 1`] = `
[
  {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
  {
    "externalType": "window",
    "request": [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
]
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`output-format-json\` should produce expected output: Asset file 'main.asset.json' should match snapshot 1`] = `"{"dependencies":["lodash"],"version":"abdb687afe609197ad7e"}"`;

exports[`DependencyExtractionWebpackPlugin Webpack \`output-format-json\` should produce expected output: External modules should match snapshot 1`] = `
[
  {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
]
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`overrides\` should produce expected output: Asset file 'main.asset.php' should match snapshot 1`] = `
"<?php return array('dependencies' => array('wp-blob', 'wp-script-handle-for-rxjs', 'wp-url'), 'version' => '2a29b245fc3d0509b5a8');
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`overrides\` should produce expected output: External modules should match snapshot 1`] = `
[
  {
    "externalType": "window",
    "request": "rxjs",
    "userRequest": "rxjs",
  },
  {
    "externalType": "window",
    "request": [
      "rxjs",
      "operators",
    ],
    "userRequest": "rxjs/operators",
  },
  {
    "externalType": "window",
    "request": [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
  {
    "externalType": "window",
    "request": [
      "wp",
      "url",
    ],
    "userRequest": "@wordpress/url",
  },
]
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`runtime-chunk-single\` should produce expected output: Asset file 'a.asset.php' should match snapshot 1`] = `
"<?php return array('dependencies' => array('wp-blob'), 'version' => '4514ed711f6c035e0887');
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`runtime-chunk-single\` should produce expected output: Asset file 'b.asset.php' should match snapshot 1`] = `
"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => '168d32b5fb42f9e5d8ce');
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`runtime-chunk-single\` should produce expected output: Asset file 'runtime.asset.php' should match snapshot 1`] = `
"<?php return array('dependencies' => array(), 'version' => 'b4058959bcd0df37cd79');
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`runtime-chunk-single\` should produce expected output: External modules should match snapshot 1`] = `
[
  {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
  {
    "externalType": "window",
    "request": [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
]
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`style-imports\` should produce expected output: Asset file 'main.asset.php' should match snapshot 1`] = `
"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => 'beb3f6752168c2983cdd');
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`style-imports\` should produce expected output: External modules should match snapshot 1`] = `
[
  {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
  {
    "externalType": "window",
    "request": [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
]
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`wordpress\` should produce expected output: Asset file 'main.asset.php' should match snapshot 1`] = `
"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => '35d057cb1c8a30094445');
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`wordpress\` should produce expected output: External modules should match snapshot 1`] = `
[
  {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
  {
    "externalType": "window",
    "request": [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
]
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`wordpress-require\` should produce expected output: Asset file 'main.asset.php' should match snapshot 1`] = `
"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => 'ed2bd4e7df46768bb3c2');
"
`;

exports[`DependencyExtractionWebpackPlugin Webpack \`wordpress-require\` should produce expected output: External modules should match snapshot 1`] = `
[
  {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
  {
    "externalType": "window",
    "request": [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
]
`;
